src/iframe/src/Iframe.js

```diff
@@ -348,8 +359,18 @@ class Iframe extends Component {
     })
   }
 
+  drawUserGraphics(state) {
+    if (window.draw) {
+      window.draw(state)
+      if (this.renderer === "framebuffer") {
+        this._pixelData.data.set(this._pixelBytes)
+        let ctx = this._canvas.getContext('2d')
+        ctx.putImageData(this._pixelData, 0, 0)
+      }
+    }
+  }
+
   componentDidMount() {
-    this.updateGlobals()
```

Comment:
I think this needs to stay here. It sets all the default globals. It's confusing and I should rename it.

Reply:
I moved this update globals to the bottom of componentDidMount because it now needs to happen after the renderer property gets set. I looked at the code between and I don't think it depends on the globals, so I figured moving it later is not a terrible strategy.

An alternative option would be to move the URLSearchParams call to the top of the function and set the renderer there before calling updateGlobals. Up to you.